### PR TITLE
Feature/add page modifier

### PR DESCRIPTION
**Pull request by groony**

### CODE REVIEW

**Comment by groony** [comment]
@ne1ro Review please

File: siteshot.coffee [hunk 1]
```diff
@@ -30,24 +30,34 @@ class SiteShot
...
-                          page.close()
-                          console.log "Finish loading #{route} and save it in #{snapPath}"
-                          callback()
+                      if (typeof config.pageModifier == 'function')
+                        config.pageModifier(page, ->
+                          generateHTML();
+                        );
+                      else
+                        generateHTML();
+                    ), config.delay || 0
```
Comment (ne1ro) [inline siteshot.coffee]: `or` is preferred than `||` by [CoffeeScript style guide](https://github.com/polarmobile/coffeescript-style-guide#miscellaneous)

File: siteshot.coffee [hunk 2]
```diff
@@ -30,24 +30,34 @@ class SiteShot
...
-                          throw err if err?
-                          page.close()
-                          console.log "Finish loading #{route} and save it in #{snapPath}"
-                          callback()
+                      if (typeof config.pageModifier == 'function')
+                        config.pageModifier(page, ->
+                          generateHTML();
+                        );
+                      else
+                        generateHTML();
```
Comment (ne1ro) [inline siteshot.coffee]: You don't need `;` in CoffeeScript.

File: siteshot.coffee [hunk 3]
```diff
@@ -30,24 +30,34 @@ class SiteShot
...
+                              console.log "Finish loading #{route} and save it in #{snapPath}"
+                              callback()
 
-                        # Write snapshot file
-                        fs.writeFile snapPath, res, (err) =>
-                          throw err if err?
-                          page.close()
-                          console.log "Finish loading #{route} and save it in #{snapPath}"
-                          callback()
+                      if (typeof config.pageModifier == 'function')
```
Comment (ne1ro) [inline siteshot.coffee]: `is` is preferred over `==` by [CoffeeScript style guide](https://github.com/polarmobile/coffeescript-style-guide#miscellaneous)

**Comment by ne1ro** [comment]
@groony reviewed

**Comment by groony** [comment]
@ne1ro Ready for re-review.

`config.js`

``` js
module.exports = {
  snapshotDir: "/var/www/APPLICATION_NAME/shadow-copy/snapshots",
  sitemap: "/var/www/APPLICATION_NAME/shadow-copy/sitemap.xml",
  delay: 5000,
  pageModifier: function(page, callback) {  page.evaluate(function() { $('meta[name=fragment]').remove() }); callback(); }
}
```

File: siteshot.coffee [hunk 4]
```diff
@@ -31,26 +31,34 @@ class SiteShot
...
+                            fs.writeFile snapPath, res, (err) =>
+                              throw err if err?
+                              page.close()
+                              console.log "Finish loading #{route} and save it in #{snapPath}"
+                              callback()
+
+                      if (typeof config.pageModifier is 'function')
+                        config.pageModifier(page, ->
+                          generateHTML()
+                        );
```
Comment (ne1ro) [inline siteshot.coffee]: Forgotten semicolon

**Comment by groony** [comment]
@ne1ro Ready)

**Comment by ne1ro** [comment]
@groony ok

**Pull request merged**